aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -77,6 +77,10 @@ Tensor arange(
     Scalar end,
     Scalar step,
     const TensorOptions& options) {
+  AT_CHECK(
+    finite(end.toDouble()) && finite(start.toDouble()),
```

Comment:
nit: std::is_finite might be more portable, but I don't know

Reply:
updated, @soumith mentioned this above as well.